src/tree/workspace.rs

```diff
@@ -274,6 +274,12 @@ impl Node for WorkspaceNode {
     fn node_do_focus(self: Rc<Self>, seat: &Rc<WlSeatGlobal>, direction: Direction) {
         if let Some(fs) = self.fullscreen.get() {
             fs.tl_into_node().node_do_focus(seat, direction);
+        } else if let Some(stacked) = self.stacked.last() {
+            if let Some(float) = (*stacked).clone().node_into_float() {
+                if let Some(child) = float.child.get() {
+                    child.node_do_focus(seat, direction);
+                }
+            }
```

Comment:
This should probably loop over all stacked nodes and pick the first one that is a float node. For example, if the topmost node is a popup, then this would not have the desired effect.